### ISSUE TITLE
Upgrade cryptography to version 3.4+

### ISSUE THREAD

**Issue by bmw** [certbot/certbot#8704]
Newer versions of cryptography require a rust compiler on systems where wheels aren't available such as some ARM systems. I think we should upgrade and get this working now rather than scrambling to do it later if we need to upgrade cryptography.

When working on this, keep in mind that we disable ARM builds on `test-` branches in https://github.com/certbot/certbot/blob/master/.azure-pipelines/templates/jobs/packaging-jobs.yml so you'll probably want to temporarily change that to ensure it works everywhere.

**Comment by ohemorange** [comment]
So, hit a slight hiccup:

```
# We pin cryptography to 3.1.1 and pyOpenSSL to 19.1.0 specifically for CentOS 7 / RHEL 7
# because these systems ship only with OpenSSL 1.0.2, and this OpenSSL version support has been
# dropped on cryptography>=3.2 and pyOpenSSL>=20.0.0.
# Using this old version of OpenSSL would break the cryptography and pyOpenSSL wheels builds.
```
(from `tests/letstest/scripts/test_sdists.sh`)

It sounds like if we require 3.4+, we have to drop support for CentOS 7 / RHEL 7, which is EOL 2024.

**Comment by ohemorange** [comment]
Presumably cryptography knows about this, and would backport a fix to 3.1?

**Comment by ohemorange** [comment]
Anyway, it's probably fine to just upgrade cryptography, and not upgrade our requirement, I had just assumed this issue was to do both.

**Comment by bmw** [comment]
Right. For this issue I was just imagining that we'd update our pinned version of `cryptography` and not change how our dependency is specified in places like `setup.py`. Because of this, I think you can continue to let `test_sdists.sh` use `sed` to pin older versions for those tests (although setting up the `sed` commands like that to begin with is the kind of thing I'd now probably argue is a poor use of our time and it's a distro problem now).

> Presumably cryptography knows about this, and would backport a fix to 3.1?

I think `cryptography` has more or less moved on and it's on distros with old version of `cryptography` to patch them although the cryptography maintainers may help with that process.

**Comment by ohemorange** [comment]
Closed in #8730.
